lib/verdict/experiment.rb

```diff
@@ -234,31 +209,27 @@ def should_store_assignment?(assignment)
   end
 
   def assignment_with_unqualified_persistence(subject_identifier, subject, context)
-    previous_assignment = fetch_assignment(subject_identifier)
+    previous_assignment = lookup(subject)
     return previous_assignment unless previous_assignment.nil?
     if subject_qualifies?(subject, context)
       group = segmenter.assign(subject_identifier, subject, context)
-      subject_assignment(subject_identifier, group, nil, group.nil?)
+      subject_assignment(subject, group, nil, group.nil?)
     else
-      subject_assignment(subject_identifier, nil, nil)
+      subject_assignment(subject, nil, nil)
     end
   end
 
   def assignment_without_unqualified_persistence(subject_identifier, subject, context)
     if subject_qualifies?(subject, context)
-      previous_assignment = fetch_assignment(subject_identifier)
+      previous_assignment = lookup(subject)
       return previous_assignment unless previous_assignment.nil?
       group = segmenter.assign(subject_identifier, subject, context)
-      subject_assignment(subject_identifier, group, nil, group.nil?)
+      subject_assignment(subject, group, nil, group.nil?)
     else
-      subject_assignment(subject_identifier, nil, nil)
+      subject_assignment(subject, nil, nil)
     end
   end
 
-  def subject_identifier(subject)
```

Comment:
sorry at some point i said we should delete this method, didn't notice it was here before.
We should keep this method here, and have `retrieve_subject_identifier` call this method. This method should be private, and `retrieve_subject_identifier` public.